library/std/src/fs.rs

```diff
@@ -366,6 +366,32 @@ impl File {
         OpenOptions::new().write(true).create(true).truncate(true).open(path.as_ref())
     }
 
+    /// Creates a new file in read-write mode.
+    ///
+    /// This function will create a file if it does not exist, or return an error if it does. This
+    /// way, if the call succeeds, the file returned is guaranteed to be new.
+    ///
+    /// This option is useful because it is atomic. Otherwise between checking whether a file
+    /// exists and creating a new one, the file may have been created by another process (a TOCTOU
+    /// race condition / attack).
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// #![feature(file_create_new)]
+    ///
+    /// use std::fs::File;
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let mut f = File::create_new("foo.txt")?;
+    ///     Ok(())
+    /// }
+    /// ```
+    #[unstable(feature = "file_create_new", issue = "none")]
+    pub fn create_new<P: AsRef<Path>>(path: P) -> io::Result<File> {
```

Comment:
nit: There is often confusion about the atomic aspect of filesystem operation. `create_new` sounds innocent, and it is not clear what the intention of the function is given its name. After all, by name one might expect the function to  "create a new file or overwrite the existing one", which is the exact opposite of what it does. Maybe `create_new_or_fail()` (or something less clunky)

Reply:
@lukaslueg I think we should be consistent with the `create_new` method of `OpenOptions`; the same argument applies to that method.

Reply:
I've expanded the documentation, including this in the summary line and not just the next paragraph, since it's a key aspect of the function's behavior.